tests/e2e/specs/backend/product-query/product-filters.test.ts

```diff
@@ -0,0 +1,271 @@
+					'Show only products on sale'
+				);
+			} );
+
+			it( 'Editor preview shows correct products corresponding to the value `Show only products on sale`', async () => {
+				const defaultCount = getFixtureProductsData().length;
+				const saleCount = getFixtureProductsData( 'sale_price' ).length;
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+				await toggleProductFilter( 'Sale status' );
+				await setCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength( saleCount );
+				await unsetCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+			} );
+
+			it( 'Works on the front end', async () => {
+				await toggleProductFilter( 'Sale status' );
+				await setCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				await canvas().waitForSelector(
+					SELECTORS.editorPreview.productsGrid
+				);
+				await saveOrPublish();
+				await shopper.block.goToBlockPage( block.name );
+				const saleCount = getFixtureProductsData( 'sale_price' ).length;
+				expect( await getFrontEndProducts() ).toHaveLength( saleCount );
+			} );
+		} );
+
+		describe( 'Stock Status', () => {
+			it( 'Stock status is enabled by default', async () => {
+				await expect( $productFiltersPanel ).toMatchElement(
+					SELECTORS.formTokenFieldLabel,
+					{ text: 'Stock status' }
+				);
+			} );
+
+			it( 'Can add and remove Stock Status filter', async () => {
+				await toggleProductFilter( 'Stock status' );
+				await expect( $productFiltersPanel ).not.toMatchElement(
+					SELECTORS.formTokenFieldLabel,
+					{ text: 'Stock status' }
+				);
+				await toggleProductFilter( 'Stock status' );
+				await expect( $productFiltersPanel ).toMatchElement(
+					SELECTORS.formTokenFieldLabel,
+					{ text: 'Stock status' }
+				);
+			} );
+
+			it( 'All statuses are enabled by default', async () => {
+				await expect( $productFiltersPanel ).toMatch( 'In stock' );
+				await expect( $productFiltersPanel ).toMatch( 'Out of stock' );
+				await expect( $productFiltersPanel ).toMatch( 'On backorder' );
+			} );
+
+			it( 'Editor preview shows all products by default', async () => {
+				const defaultCount = getFixtureProductsData().length;
+
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+			} );
+
+			/**
+			 * Skipping this test for now as Product Query doesn't show correct set of products based on stock status.
+			 *
+			 * @see https://github.com/woocommerce/woocommerce-blocks/pull/7682
+			 */
+			it.skip( 'Editor preview shows correct products that has enabled stock statuses', async () => {
+				const $$tokenRemoveButtons = await $productFiltersPanel.$$(
+					SELECTORS.tokenRemoveButton
+				);
+				for ( const $el of $$tokenRemoveButtons ) {
+					await $el.click();
+				}
```

Comment:
@sunyatasattva What do you think about prefixing variables containing an array of DOM elements/nodes with `$$` like this? At a glance, did it raise any questions in your mind?

Reply:
I've never done it but I think it makes sense.